pkg/irreparabledb/irreparabledb.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package irreparabledb
+
+import (
+	"context"
+	"time"
+
+	"storj.io/storj/internal/migrate"
+	dbx "storj.io/storj/pkg/irreparabledb/dbx"
+)
+
+// Database implements the irreparable RPC service
+type Database struct {
+	db *dbx.DB
+}
+
+// RemoteSegmentInfo is info about a single entry stored in the irreparable db
+type RemoteSegmentInfo struct {
+	SegmentPath        []byte
+	SegmentDetail      []byte
+	LostPiecesCount    int64
+	RepairUnixSec      int64
+	RepairAttemptCount int64
+}
+
+// New creates instance of Server
+func New(source string) (*Database, error) {
```

Comment:
It should also work with postgres.

Reply:
the db.Exec() take care of it ... I will double check

Reply:
You are currently always creating a sqlite driver.

Reply:
couldn't get a single query/transaction, that could work with both databases. Will have to seek databases expert's help in getting a single transaction/query. This could still be optimized.